### PR TITLE
Add no_syslog option in order to prevent ansible syslog its actions on the remote host

### PR DESCRIPTION
**Pull request by lberruti**
Add no_syslog option in order to prevent ansible syslog its actions
on the remote host.

Syslog on remote hosts can be turned off by:
- no_syslog option inside ansible.cfg.
- -n --no_syslog command line options (ansible and ansible-playbook).

### CODE REVIEW

**Comment by bcoca** [comment]
so we already have a no_log tag that seems to overlap with this

**Comment by lberruti** [comment]
The behaviours are different:
- no_log hides contents of _sensible_ data on control machine (see http://docs.ansible.com/faq.html#how-do-i-keep-secret-data-in-my-playbook), i.e.

```
# control machine:
TASK: [command sleep 1 chdir=/tmp] ********************************************
ok: [test001] => {"censored": "results hidden due to no_log parameter"}
```
- no_syslog doesn't touch information visualized on the control machine, but it removes all syslog lines on the target machine. i.e no more lines lines like

```
# target / remote machine:
Jul  6 15:37:22 test001 ansible-command: Invoked with creates=None executable=None chdir=/tmp args=sleep 1 removes=None NO_LOG=None shell=False warn=True
```

**Comment by bcoca** [comment]
no_log also prevents logging to syslog of that sensitive data, if used at the task level it should also prevent all syslog entries for that task on the remote machine.

File: lib/ansible/module_utils/basic.py [hunk 1]
```diff
@@ -399,7 +401,7 @@ def __init__(self, argument_spec, bypass_checks=False, no_log=False,
             self._check_required_if(required_if)
 
         self._set_defaults(pre=False)
-        if not self.no_log:
+        if not (self.no_log or self.no_syslog):
```
Comment (bcoca) [inline lib/ansible/module_utils/basic.py:404]: here where you prevent logging with no_syslog, no_log already did the same

**Comment by lberruti** [comment]
It seems that no_log has been changed from stable to devel:

```
# stable-1.9
TASK: [secret task] ***********************************************************
changed: [deb7.inter.nos] => {"censored": "results hidden due to no_log parameter", "changed": true, "rc": 0}
# devel
TASK [secret task] **************************************************************
changed: [deb7.inter.nos] => {"changed": true, "end": "2015-07-06 17:35:58.424702", "stdout": "value=secret_to_hide", "cmd": "echo value=secret_to_hide", "start": "2015-07-06 17:35:58.423016", "delta": "0:00:00.001686", "stderr": "", "rc": 0, "invocation": {"module_name": "command", "module_args": {"_raw_params": "echo value=secret_to_hide", "_ansible_no_log": true, "_uses_shell": true}}, "stdout_lines": ["value=secret_to_hide"], "warnings": []}
```

Which one is the "official" behaviour?

The purpose of this pr is to turn off all kind of "ansible syslog" on remote machines (even syslog of "setup tasks") via ansible.cfg  (or via command line switch level)

I have added a new option instead to modify no_log because:
1. My understanding of no_log is: "used to censor sensible data (also) on control machine"
2. I want to leave untouched output on controlo machine

**Comment by bcoca** [comment]
1) yes but s/sensible/sensitive/
2) no_syslog then might want to turn into no_remote_syslog

**Comment by lberruti** [comment]
1) oops! Sorry :)
2) ok to rename: I will do it tomorrow

**Comment by bcoca** [comment]
let me bring this up with rest of core first

**Comment by amenonsen** [comment]
Any verdict?

**Comment by lberruti** [comment]
I can make it works again
Please let me known if the core prefer no_remote_syslog or no_syslog

**Comment by amenonsen** [comment]
I think `no_syslog` is fine; we don't syslog locally anyway, so…

**Comment by lberruti** [comment]
I rebased and submitted this as #12140, this PR can now be closed.

**Comment by amenonsen** [comment]
@lberruti it's your own PR, you can close it yourself. :-)

**Comment by lberruti** [comment]
@amenonsen ok :)

**Pull request closed without merging**